blog/predictive-analytics.qmd

```diff
@@ -0,0 +1,131 @@
+
+::: {.callout-note appearance="simple"}
+## The objective of this model is to classify whether or not a patient's length of stay will be \< or \>= 7 days, rather than predicting the exact number of days!
+:::
+
+## Data Set Preparation & Variable Selection
+
+The variables used in the data set to train the model were selected as it was thought they would be useful as indicators for a patient's length of stay. These included age, elective / non-elective admission, whether the patient was admitted from a nursing home or not, theatre records, the patient's sepsis status and the presence of particular comorbidities (that is previous diagnosis of dementia, COPD, heart failure and diabetes).
+
+To allow the model to be trained on this data, all of the categorical variables had to be re-categorised using "one hot encoding", where the values of each of these variables would then be represented in a separate column identified with a 1 or a 0. This alteration allows the input into the model to be numeric. As only categorical variables needed to be encoded, age did not need to be altered.
+
+::: {.callout-note appearance="simple"}
+## For example Instead of the comorbidity columns stating "Yes" or "No" for each condition, this would be represented with separate columns. These columns would state whether the patient had each relevant diagnosis with a 1 for true and 0 for false.
+:::
+
+## Preparation Of Model
+
+The R script used to prepare and evaluate this model utilised the following packages, "stats", "dplyr" & "randomForest". Initially, the data set was loaded into R from a CSV file and the columns checked (for example formatting and column names). In this example, the CSV file is referred to as "DATASET". Following this, the data set is split into a testing section and a training section. For this I opted for a split of 70% training and 30% testing. Then, the random forest model is built, specifying that the field "Length.Of.Stay" is the variable I am looking to predict and that I want to make this prediction based on all of the other variables available in the data set. Lastly, the model accuracy is evaluated through a confusion matrix. This model presents an accuracy of 85.11%, when evaluating the predictions made from the testing data set.
+
+::: {.callout-caution appearance="simple"}
+## In this case, set.seed(x) is important in order to make your results reproducible. If a different seed number was used to re-train the model, your final accuracy score may differ, even though the same data set was used for training.
+:::
+
+```{r eval=FALSE}
+# load packages
+library(stats)
```

Comment:
Minor point, but {stats} is part of base R and is attached on startup; the `library()` call is unneeded.